### PR TITLE
Fixes #284: adding support for sub project api docs

### LINKED ISSUE #284: add api docs support to sub project documentation generation
### This is a
- [ ] Bug report
- [x] Feature request

### Current behavior

sub projects are cloned and documentation is generated using sphinx
### Expected behaviour

Each project has it's own configuration like swagger, api blueprint, javadocs, so add the support for api documentation for each project and combine as a single doucmentation. 

### Steps to reproduce
1.
2.
3.

### PR DESCRIPTION
**Pull request by sch00lb0y**
## Description

added support for generating api docs for sub project
## Related Issue

close #284 
## Motivation and Context

Right now yaydoc generating api docs only for single reposiotry. In subprojects each project need diffreent api doc. This will enable us to generate api docs for sub project also.
## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by sch00lb0y** [comment]
Now I'm generating documentation separately for each repository and adding into the main repository generated folder. @pri22296 @imujjwal96 please check the current implementation

**Comment by sch00lb0y** [comment]
@pri22296 @imujjwal96  please check the deployment http://demo08.herokuapp.com/

**Comment by sch00lb0y** [comment]
@imujjwal96 @pri22296 please review this

**Comment by imujjwal96** [comment]
Rebase.

**Comment by sch00lb0y** [comment]
@imujjwal96 updated

File: generate.sh [hunk 1]
```diff
@@ -98,97 +98,110 @@ fi
 
 mkdir yaydoctemp
 BUILD_DIR=$(pwd)/yaydoctemp
+TEMP_NAME=yaydoctemp
+function generate() {
```
Comment (imujjwal96) [inline generate.sh]: is the `()` required?

File: generate.sh [hunk 2]
```diff
@@ -98,97 +98,110 @@ fi
...
 
-cd $ROOT_DIR
+  cd $1
```
Comment (imujjwal96) [inline generate.sh]: what's the difference in style with line 109?

Comment (sch00lb0y) [inline generate.sh]: updated

File: generate.sh (hunk 1 above)
Comment (sch00lb0y) [inline generate.sh]: updated

File: generate.sh [hunk 3]
```diff
@@ -98,97 +98,110 @@ fi
 
 mkdir yaydoctemp
 BUILD_DIR=$(pwd)/yaydoctemp
+TEMP_NAME=yaydoctemp
+function generate {
 
-cd $ROOT_DIR
+  cd $1
```
Comment (imujjwal96) [inline generate.sh]: i think we should use all the variables like `${variable}` instead of `$variable`
@pri22296  suggestions?

Comment (pri22296) [inline generate.sh]: Store the positional args in named variables or add comment explaining each positional args

File: modules/scripts/genindex.py [hunk 4]
```diff
@@ -113,9 +122,12 @@ def main():
                         help='Path of java source files for Javadoc')
     parser.add_argument('-d', '--sub-docpaths', default='',
                         help='Comma seperated docpaths for subprojects')
+    parser.add_argument('-b', '--build-directory', default='',
+                        help='Comma seperated docpaths for subprojects')
```
Comment (pri22296) [inline modules/scripts/genindex.py]: update this help message

File: modules/scripts/genindex.py [hunk 5]
```diff
@@ -96,14 +96,23 @@ def get_index(root, subprojects, sub_docpaths, javadoc):
...
+    if build_dir != "":
+        index = add_subproject(index, build_dir)
     return '\n\n'.join(index) + '\n'
 
 def add_javadoc(index):
     javadoc = ["API Documentation", "\n", "=================", "\n",
      "* `javadoc <./javadoc>`_"]
     return index + javadoc
 
+def add_subproject(index, build_dir):
```
Comment (pri22296) [inline modules/scripts/genindex.py]: Why is this needed. Sub projects were already being included in the generated index

File: multiple_generate.sh [hunk 6]
```diff
@@ -4,17 +4,47 @@ function cloneSubProject() {
...
+  export AUTOAPI_PYTHON=""
+  export AUTOAPI_PYTHON_PATH=""
+  export AUTOAPI_JAVA=""
+  export AUTOAPI_JAVA_PATH=""
+  export MOCK_MODULES=""
+  export DOCURL=""
+  export HEROKU_APP_NAME=""
+  export SWAGGER_SPEC_URL=""
+  export SWAGGER_UI=""
+  export JAVADOC_PATH=""
```
Comment (pri22296) [inline multiple_generate.sh]: Why do you need to initialize these variables? Please remove them.

File: multiple_generate.sh [hunk 7]
```diff
@@ -4,17 +4,47 @@ function cloneSubProject() {
...
+
+  ENVVARS="$(python ${BASE}/modules/scripts/config.py "${USERNAME}" "${REPONAME}")"
```
Comment (pri22296) [inline multiple_generate.sh]: Please ensure that multiple_generate only contains code which is exclusively needed for sub projects. I see no reason why reading config should be in multipl_generate

File: generate.sh (hunk 3 above)
Comment (pri22296) [inline generate.sh]: Let's be consistent. Use `${variable}` instead of `$variable` and also `[ ]` and `[[ ]]` are both used with `if`. Please only use one way in the entire script.

File: multiple_generate.sh (hunk 7 above)
Comment (sch00lb0y) [inline multiple_generate.sh]: @pri22296 each project having a different configuration. let say `open-event-server` need api blueprint and `open-event-android` need Javadoc. so users have to specify the advanced configuration in each project and based on that documentation will be built.

File: multiple_generate.sh (hunk 6 above)
Comment (sch00lb0y) [inline multiple_generate.sh]: `config.py` is not overriding if the variable already presents so I'm re initializing. (variables are intialized on the first run)

File: multiple_generate.sh (hunk 7 above)
Comment (pri22296) [inline multiple_generate.sh]: Does that mean that every sub project will have it's own `.yaydoc.yml`?

File: modules/scripts/genindex.py (hunk 5 above)
Comment (sch00lb0y) [inline modules/scripts/genindex.py]: @pri22296 now the documentation is not generated as a single project like before. now documentation is generated separately according to `.yaydoc.yaml` specified in each project. so I added a function which walks the folder and links it to the index.rst.

Comment (pri22296) [inline modules/scripts/genindex.py]: Please update the description of the PR with all the changes and assumptions. Major changes should be described in detail.

File: multiple_generate.sh (hunk 7 above)
Comment (sch00lb0y) [inline multiple_generate.sh]: yes, every project should have `.yaydoc.yaml` if it needs advanced configuration

Comment (pri22296) [inline multiple_generate.sh]: Ok. Then I don't think this is a proper approach. This should be done from the webUI's end and not from the script's end. It would be preferable if from the webUI you just call the script in a loop and then call a seperate script that just aggregates the html from all the previews into a single folder.
@imujjwal96 Your views.

File: modules/scripts/genindex.py (hunk 5 above)
Comment (imujjwal96) [inline modules/scripts/genindex.py]: What if i do not specify .yaydoc.yml in those subprojects?

Comment (sch00lb0y) [inline modules/scripts/genindex.py]: It'll take the default config from `config.py`

File: multiple_generate.sh (hunk 7 above)
Comment (sch00lb0y) [inline multiple_generate.sh]: if the user didn't specify the sub project in webui but he specified in the main project `.yaydoc.yaml` what we can do about this? please consider this situation also

**Comment by imujjwal96** [comment]
Does these changes mean that i can define subprojects for each subproject of a master project?

File: multiple_generate.sh (hunk 7 above)
Comment (imujjwal96) [inline multiple_generate.sh]: Which of these two approaches will generate docs for the projects concurrently?

Comment (pri22296) [inline multiple_generate.sh]: The current model is a pull only model. Since you are also planning to build the project whenever any sub-project is updated as per #297 so the model would have to anyways be scrapped. What i propose is that remove `subprojects` from the `.yaydoc.yml` option and instead subprojects would have to be registered from the webUI itself and I don't see any harm in making `.yaydoc.yml` compulsory for subprojects too.

Comment (sch00lb0y) [inline multiple_generate.sh]: your suggestion for removal of `subproject` sounds good then I'll change to create documentation from sub project from WEBUI . Actually, #297 won't scrape the current implementation. we just register web hook to all the sub project if it triggers we generate documentation based on the config we saved in the DB. merging of this PR won't be a problem.

**Comment by sch00lb0y** [comment]
@imujjwal96 you can't because subproject's sub project config is not override

**Comment by sch00lb0y** [comment]
@pri22296 @imujjwal96 I rewrote the implementation. deployment link: http://demo08.herokuapp.com/.
please check

**Comment by sch00lb0y** [comment]
@pri22296 @imujjwal96 please review this

File: backend/generator.js [hunk 8]
```diff
@@ -11,57 +11,121 @@ exports.executeScript = function (socket, formData, callback) {
...
-        })
+          message: `Process exited with code : ${failedData.code}`
+        });
       }
     }
   });
-  return true;
 };
+
+exports.retrieveLogs = function (socket, data) {
```
Comment (imujjwal96) [inline backend/generator.js]: i guess this was moved to `output.js`

Comment (sch00lb0y) [inline backend/generator.js]: While resolving conflicts I forgot to remove :sweat_smile:

**Comment by sch00lb0y** [comment]
@imujjwal96 I have updated. If any PR merge it making difficult to remove the conflicts for this PR. so @pri22296 please review this PR.

File: backend/generator.js [hunk 9]
```diff
@@ -11,57 +11,112 @@ exports.executeScript = function (socket, formData, callback) {
...
+          message: `Process exited with code : ${failedData.code}`
+        });
       }
     }
   });
-  return true;
 };
+
+var generate = function (socket, data) {
+  return new Promise(function(resolve, reject) {
```
Comment (imujjwal96) [inline backend/generator.js]: What is the use of Promise here?

File: backend/generator.js [hunk 10]
```diff
@@ -11,57 +11,112 @@ exports.executeScript = function (socket, formData, callback) {
...
-  var process = spawn("./generate.sh", args);
-
-  socketHandler.handleLineOutput(socket, process, 'logs', 4);
-  socketHandler.handleLineError(socket, process, 'err-logs');
+  var buildId = uuidV4();
+  var subProjectNames = [];
+  subProjectUrl.forEach(function (x) {
+     subProjectNames.push(x.split("/")[4].split(".")[0]);
+  });
+  var mainProjetData = {
```
Comment (imujjwal96) [inline backend/generator.js]: spell check

Comment (imujjwal96) [inline backend/generator.js]: Why create a separate variable here? Why not use `data` as the variable?

File: backend/generator.js [hunk 11]
```diff
@@ -11,57 +11,112 @@ exports.executeScript = function (socket, formData, callback) {
...
+  generatedData.then(function (result) {
+    var flag = true;
+    var failedData;
+    result.forEach(function (x) {
+      if (x.code !== 0) {
+        flag = false;
+        failedData = x;
+      }
+    });
+    if (flag) {
```
Comment (imujjwal96) [inline backend/generator.js]: What does this flag define?

File: backend/generator.js (hunk 9 above)
Comment (sch00lb0y) [inline backend/generator.js]: While generating documentation for sub project. I have run the generate function concurrently. Promise has a method `Promise.all`. so I'll pass an array of promise and it'll give the resolved result in an array. even the code look clean.

File: backend/generator.js (hunk 10 above)
Comment (sch00lb0y) [inline backend/generator.js]: I'm using data as an array to push all the projects data

File: backend/generator.js (hunk 11 above)
Comment (sch00lb0y) [inline backend/generator.js]: flag is used to check any of the project builds is failing or not. if it fails flag became false and error will be emitted in socket

File: backend/generator.js (hunk 9 above)
Comment (imujjwal96) [inline backend/generator.js]: Look for https://github.com/caolan/async It's use has been recommended in nodejs backend. Promises are mainly used in the front end

Comment (sch00lb0y) [inline backend/generator.js]: I don't find any harm in using promises and also it is supported natively by nodejs. also there no thing such promise is mainly used in the frontend. please check [Ghost](https://github.com/TryGhost/Ghost) They using Promise throughout the backend.

File: backend/generator.js (hunk 10 above)
Comment (sch00lb0y) [inline backend/generator.js]: I have updated the spell.

File: modules/scripts/genindex.py [hunk 12]
```diff
@@ -80,22 +77,16 @@ def get_index(root, subprojects, sub_docpaths, javadoc):
     for (dirpath, dirnames, filenames) in os.walk(os.curdir):
         _ignore_files(dirpath, filenames, ['README.rst', 'README.md'], root)
         _ignore_dirs(dirpath, dirnames, ['yaydoctemp', 'yaydocclone', '.git'])
-        _ignore_dirs(dirpath, dirnames, ['source'] + subproject_dirs, os.curdir)
```
Comment (pri22296) [inline modules/scripts/genindex.py]: Don't remove this entirely. Replace with `_ignore_dirs(dirpath, dirnames, ['source'], os.curdir)`

File: generate.sh [hunk 13]
```diff
@@ -177,14 +175,21 @@ if [ -n "$SWAGGER_SPEC_URL" ]; then
...
-if [[  -n "$JAVADOC_PATH" ]]; then
-  javadoc -sourcepath $ROOT_DIR/$JAVADOC_PATH -d $BUILD_DIR/_build/html/javadoc -subpackages . >> ${LOGFILE} 2>>${LOGFILE}
+if [ -n "${JAVADOC_PATH}" ]; then
+  javadoc -sourcepath ${ROOT_DIR}/${JAVADOC_PATH} -d ${BUILD_DIR}/_build/html/javadoc -subpackages . >> ${LOGFILE} 2>>${LOGFILE}
 fi
 
 print_log "Setting up documentation for Download and Preview\n"
-mv $BUILD_DIR/_build/html $ROOT_DIR/../${UNIQUEID}_preview && cd $_/../
+
+if [ -n "${SUBPROJECT}" ]; then
```
Comment (pri22296) [inline generate.sh]: Better to compare it with `true`

**Comment by sch00lb0y** [comment]
@pri22296 updated

File: README.md [hunk 14]
```diff
@@ -9,6 +9,7 @@
 
 ## Prerequisites
 - Create a directory in your repository containing all the markup files along with an `index.rst` file which contains [toctrees](http://www.sphinx-doc.org/en/stable/markup/toctree.html) to link the various documents. You can also provide the index in markdown format although then you won't be able to include other documents such as the `README` into them.
+- If you using the manual index for sub project documentation generation, link to your subprojects using ``` `repository name <./repository name>`_``` in the index.
```
Comment (pri22296) [inline README.md]: Create a different section for this because links to javadocs, swagger would also go there

Comment (sch00lb0y) [inline README.md]: can I name the title as `Advance configuration for the manual index`? or
if you have anything in mind. please give your input

File: modules/scripts/genindex.py [hunk 15]
```diff
@@ -104,27 +96,25 @@ def add_javadoc(index):
...
+        subprojects_toc_tree.append("\n")
+        subprojects_toc_tree.append("* `{0} <./{0}>`_".format(name))
+    return index + subprojects_toc_tree
+
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('root', help='Path to the root of the Repository')
     parser.add_argument('-s', '--subprojects', default='',
-                        help='Comma seperated subprojects')
+                        help='Comma seperated subprojects name')
```
Comment (pri22296) [inline modules/scripts/genindex.py]: subproject names

**Comment by pri22296** [comment]
@sch00lb0y Can you explain the use of BUILDID and UNIQUEID?

File: modules/scripts/genindex.py (hunk 15 above)
Comment (sch00lb0y) [inline modules/scripts/genindex.py]: Actually, it is comma separated sub projects name only.

**Comment by sch00lb0y** [comment]
@pri22296 `BUILDID` is the unique id for the documentation build. `UNIQUEID` is the unique id for the concurrent documentation generation

File: README.md (hunk 14 above)
Comment (pri22296) [inline README.md]: No need for explicitly saying `manual index`. I think it's implied. Other than that sounds fine.

File: modules/scripts/genindex.py (hunk 15 above)
Comment (pri22296) [inline modules/scripts/genindex.py]: Yeah I understand that. What I'm saying is that the `s` should be with `name` instead of `subproject`.

**Comment by pri22296** [comment]
@sch00lb0y Thanks for the explanation. I had difficulty following what is used for what hence asked. Might as well add a comment in the code.

File: README.md (hunk 14 above)
Comment (sch00lb0y) [inline README.md]: updated

File: modules/scripts/genindex.py (hunk 15 above)
Comment (sch00lb0y) [inline modules/scripts/genindex.py]: updated

**Comment by pri22296** [comment]
@sch00lb0y I just noticed that the search doesn't work for subprojects. It used to work with the earlier implementation. Any idea why that happens?

**Comment by sch00lb0y** [comment]
In past implementation, we generated documentation as a single project but in this implementation, we generating documentation for each project separately and moving to a single folder. but the search will work for sub project after you go to the sub project documentation

**Comment by imujjwal96** [comment]
The point of aggregating sub projects to a single project is to make that as a single entity.
If we cannot search for something inside the sub projects directory, then it doesn't make any sense in having such aggregation

**Comment by sch00lb0y** [comment]
@imujjwal96 I totally agree with your view but then this PR is really important for `open-event-orga-server` and `open-event-android` for api blueprint and Javadoc. please raise this as a separate issue. I'll work on it. In order to make it work, I have to re write the whole implementation and also it takes more time. we can't delay the documentation deployment of the open-event project as we meeting our deadline.

**Comment by pri22296** [comment]
@sch00lb0y If the entire implementation would again need to be changed, I don't get the point of this PR. Anyways can you please explain exactly in detail why can't we generate apidocs for subprojects now?

**Comment by sch00lb0y** [comment]
In Our past implementation. we not creating documentation for sub projects we just copying files of sub project into a folder and creating a single documentation. we just flatting instead of creating sub documentation for sub project. our .yaydoc.yml works on main repository only. In FOSSASIA, all the main repository containing the information of the projects, not containing code, whatever we improved API docs will be unusable on sub projects.  The main reason we can't create API doc on the previous implementation is toc tree.  if we make some hack now also. it can't be managed, `genindex.py` will keep on growing only and also we keep on increasing the complexity, which can't be managed and also we can't avoid the use of `genindex.py`.we all know automatic index.rst is the hard dependency. if we ask users to generate the manual index. that won't be a viable solution. The main glitch in new implementation is searching sub project documentations from any other sub project documentation example: searching for `open-event-android` docs from `open-event-webapp` docs (logically user won't search for other documentation from staying on other documentation) @pri22296 @imujjwal96 please input your view on this problem.right now, solving this issue is important for integrating yaydoc to `open-event` project.

**Comment by pri22296** [comment]
If the search doesn't work for sub projects from the main page, I don't think there's any advantage of sub projects. The idea of sub project is that docs from multiple repositories would be combined to form a single unit. Also, I would again repeat that we should not try to avoid the use of a manual index.rst.

Though I agree with you that the current implementation isn't suitable for API-documentation where each project has different needs but from what I remember this would be the third time sub projects would need to be reimplemented. I think a lot of useful time has been wasted due to a lack of detailed discussion on this topic. You seem to rush to implement stuff which sometimes leads to a situation like this where some cases aren't handled properly and it leads to reimplementation which wastes your own important time. Let us ensure that such situation don't happen again. @imujjwal96 @sch00lb0y Please give your ideas on how can sub projects properly be implemented and this time let's ensure that all edge cases are handled.

**Comment by sch00lb0y** [comment]
@pri22296 @imujjwal96 in our current implementation. the documentation folder will come like this 

->mainproject docs
-> subproject folder
---> subproject docs
so, we'll run the api docs after documentation generation and we'll move the api docs to specific sub project folder and we'll ask the user to give index file.  the search feature also work since we generating documentation as a single folder.

**Comment by pri22296** [comment]
@sch00lb0y That sounds like a good idea. I think that should work

**Comment by sch00lb0y** [comment]
@imujjwal96 @pri22296 I'm working on the new implementation

**Comment by pri22296** [comment]
@sch00lb0y How will the user provide inputs for what apidoc to use for which subproject? currently we can only generate a single javadoc or swagger api.

**Comment by sch00lb0y** [comment]
@pri22296 `.yaydoc.yml` has to defined in each repository which need api docs

**Comment by sch00lb0y** [comment]
@pri22296 @imujjwal96 I have made the new implementation. here is the deployment please check
http://demo08.herokuapp.com/

**Comment by sch00lb0y** [comment]
@imujjwal96 @pri22296 please check this

File: generate.sh [hunk 16]
```diff
@@ -18,6 +18,35 @@ do
...
+  eval ${ENVVARS}
+
+  if [ -n "${SWAGGER_SPEC_URL}" ]; then
+    if [ "${SWAGGER_UI}" == "swagger" ]; then
+      STABLE_SWAGGER=$(curl -s https://api.github.com/repos/swagger-api/swagger-ui/releases/latest  | grep tarball_url | cut -d '"' -f 4) >/dev/null 2>&1
+      wget -O swagger.tar.gz ${STABLE_SWAGGER} >/dev/null 2>&1
+      mkdir swagger
+      tar -xf swagger.tar.gz --strip-components=1 -C swagger >/dev/null 2>&1
+      cp -r swagger/dist ${2}
+      sed -i -e "s|http://petstore.swagger.io/v2/swagger.json|${APIDOCS_URL}|g" ${2}/index.html
```
Comment (pri22296) [inline generate.sh]: From where does `$APIDOCS_URL` come from? it should be `$SWAGGER_SPEC_URL`.

Comment (sch00lb0y) [inline generate.sh]: updated

File: multiple_api_docs.sh [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+
+SUBPROJECT_DIRS_ARRAY=()
+
+for ((i=0;i<${#SUBPROJECT_URLS_ARRAY[@]};i++)); do
+  SUB_URL="${SUBPROJECT_URLS_ARRAY[$i]}"
+  SUB_DOCPATH="${SUBPROJECT_DOCPATHS_ARRAY[$i]}"
+  SUB_URL_SPLIT=(${SUB_URL//// })
+  SUB_REPONAME=(${SUB_URL_SPLIT[3]//./ })
+  generateSubAPIDocs ${SUB_REPONAME} ${SUB_DOCPATH}
+done
```
Comment (pri22296) [inline multiple_api_docs.sh]: Why is this repeated in both multiple_generate.sh and here?

Comment (sch00lb0y) [inline multiple_api_docs.sh]: `multiple_generate.sh` loop for cloning sub projects and this loop for generating api docs after sphinx doc generation

Comment (pri22296) [inline multiple_api_docs.sh]: By repeated I meant, It's mostly similar. You are doing many things again. Would it be possible if this repetition can be avoided?

Comment (sch00lb0y) [inline multiple_api_docs.sh]: loop can't be avoided because API docs have to be generated after documentation generation.

File: multiple_api_docs.sh [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+
+# shellcheck disable=SC2034
+SUBPROJECT_DIRS=$(IFS=, ; echo "${SUBPROJECT_DIRS_ARRAY[*]}")
```
Comment (pri22296) [inline multiple_api_docs.sh]: `SUBPROJECT_DIRS_ARRAY` is empty. You didn't push into it.

File: multiple_api_docs.sh (hunk 17 above)
Comment (pri22296) [inline multiple_api_docs.sh]: I'm not talking about the loop. What I'm saying is that the entire block from `IFS ...` in the two script is same except the function call. It would be better if that code repetition can be avoided.

File: multiple_api_docs.sh (hunk 18 above)
Comment (sch00lb0y) [inline multiple_api_docs.sh]: I have updated

File: multiple_api_docs.sh (hunk 17 above)
Comment (sch00lb0y) [inline multiple_api_docs.sh]: made changes please check

**Comment by sch00lb0y** [comment]
@pri22296 @imujjwal96 I have udpated

**Comment by imujjwal96** [comment]
Resolve conflicts

**Comment by sch00lb0y** [comment]
@imujjwal96  conflicts resolved

**Comment by pri22296** [comment]
@sch00lb0y Application error with your deployment

**Comment by sch00lb0y** [comment]
@pri22296 I was testing my work. I have updated now. please check now

**Comment by pri22296** [comment]
@sch00lb0y How do i specify the doc path for sub-projects?

**Comment by sch00lb0y** [comment]
@pri22296 doc path for subprojects will be specified on main project `.yaydoc.yml` as our old implementation. if users want API docs on sub project, then the user has to specify the API docs config in sub projects.

**Comment by pri22296** [comment]
Oh, cool. I was under the impression you removed those parameters.

**Comment by imujjwal96** [comment]
Everytime I plan to review this PR, i am unable to do so because of the callback url pointing to localhost.

**Comment by sch00lb0y** [comment]
@imujjwal96 updated,

**Comment by sch00lb0y** [comment]
@pri22296 @imujjwal96 please review this PR. :sweat_smile: :sweat_smile:

**Comment by imujjwal96** [comment]
@sch00lb0y How can i test this? What changes do i need to make, and where?

**Comment by sch00lb0y** [comment]
@imujjwal96 main repo : http://github.com/sch00lb0y/tempCheck 
sub repo: http://github.com/sch00lb0y/susi_server (I have added config for javadoc in `.yaydoc.yml`)
just do our normal procedure generating docs in WEBUI. you can see the javadoc generation in subdoc also

**Comment by imujjwal96** [comment]
@sch00lb0y From which branch does it generate the subdocs' apidocs?

**Comment by sch00lb0y** [comment]
@imujjwal96 GitHub default branch. let say susi_server default branch is development so it clones development

**Comment by pri22296** [comment]
@sch00lb0y Tested this with,

- https://github.com/pri22296/mainrepo
- https://github.com/pri22296/subrepo1
- https://github.com/pri22296/subrepo2

The swagger docs have been generated but they use the default swagger.json file. Not the specified one.

**Comment by sch00lb0y** [comment]
@pri22296 It is working check this http://demo08.herokuapp.com/preview/rbalajis25@gmail.com/ba9a6935-765f-43c4-8e40-9d37d4450941_preview/subrepo1/apidocs/ but showing invalid spec

**Comment by sch00lb0y** [comment]
@imujjwal96 please check this
